gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSampling.java

```diff
@@ -134,16 +129,6 @@ public LogicalGraph sample(LogicalGraph graph) {
       .filter(new PageRankResultVertexFilter(
         threshold, sampleGreaterThanThreshold, keepVerticesIfSameScore));
 
-    DataSet<Edge> newEdges = graph.getEdges()
-      .join(scaledVertices)
-      .where(new SourceId<>()).equalTo(new Id<>())
-      .with(new LeftSide<>())
-      .join(scaledVertices)
-      .where(new TargetId<>()).equalTo(new Id<>())
-      .with(new LeftSide<>());
-
-    graph = graph.getFactory().fromDataSets(scaledVertices, newEdges);
-
-    return graph;
+    return graph.getFactory().fromDataSets(scaledVertices, graph.getEdges()).verify();
```

Comment:
Seems a bit like an overhead to use verify at this point. First, the original join is smaller than the one used in verify. Second the original method creates an already smaller graph in the first place (fewer assignments of the new graph head to the edges).  I prefer the original method.

Reply:
It uses the exact same joins as before. Only `fromDataSets` is redundant, but that passes the graph head without any id assignments anyway.